### DISCUSSION TITLE
The character's ID changed after a brief loss

### DISCUSSION THREAD

**Discussion by DreamerYinYu** [roboflow/supervision#1721] [Q&A]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Question

Hello, I am using supervision ByteTrack, in the video, the male ID is 2 and the female ID is 3. However, when the female continues to move forward and the ID is lost, when it is detected again, the female ID is no longer 3, and the male and female IDs alternate. What should I do?Thank you

Male ID: 2
Girl ID: 3
![1](https://github.com/user-attachments/assets/a6143eff-381d-4aaf-82e4-db49cd0c4af4)

![2](https://github.com/user-attachments/assets/dacf22d4-a364-4d46-8c65-9ab47c8af72e)

Male ID: 3
Female ID: 2
![3](https://github.com/user-attachments/assets/475f6e81-9568-49fc-bc15-fb278a6014e0)

**Comment by DreamerYinYu** [comment]
code：


import numpy
import supervision
from ultralytics import YOLO

model = YOLO("yolo11n.pt")
tracker = supervision.ByteTrack(track_activation_threshold=0.8,minimum_matching_threshold=0.9)
box_annotator = supervision.BoxAnnotator()
label_annotator = supervision.LabelAnnotator(text_scale=1.5)
trace_annotator = supervision.TraceAnnotator()

def callback(frame: numpy.ndarray, _: int) -> numpy.ndarray:
    results = model(frame)[0]
    detections = supervision.Detections.from_ultralytics(results)
    detections = tracker.update_with_detections(detections)

    labels = [
        f"#{tracker_id} {results.names[class_id]}"
        for class_id, tracker_id
        in zip(detections.class_id, detections.tracker_id)
    ]

    annotated_frame = box_annotator.annotate(
        frame.copy(), detections=detections)
    annotated_frame = label_annotator.annotate(
        annotated_frame, detections=detections, labels=labels)
    return trace_annotator.annotate(
        annotated_frame, detections=detections)

supervision.process_video(
    source_path="videos/07.mp4",
    target_path="result.mp4",
    callback=callback
)

**Comment by LinasKo** [comment]
Hi @DreamerYinYu 👋

This is a flaw of the `ByteTrack` algorithm, unfortunately. It does not look at how detections look - only detection size, position and speed.

One non-guaranteed solution is to play around with the [minimum matching threshold](https://supervision.roboflow.com/latest/trackers/#supervision.tracker.byte_tracker.core.ByteTrack(minimum_matching_threshold)).

If not, ReID and BotTrack are solutions that could help, but we don't have them implemented yet.

Lastly, SAM2 is absolutely astonishing at tracking, but you typically need prerecorded videos.
https://blog.roboflow.com/sam-2-video-segmentation/

Edit: SAM can be read into supervision with `from_sam`, as mentioned in [this article](https://roboflow.com/how-to-plot/sam).

**Reply by DreamerYinYu** [reply]
Okay, thank you for your reply. How to use Sam-2 for multi-target tracking in supervision? thank you.

**Reply by LinasKo** [reply]
There's no end-to-end guide at the moment. The links I provided should help out, if you'd like to try and piece a solution together.
